### PR TITLE
Feature/177 create applicant dashboard

### PR DESCRIPTION
**Pull request by todd-dembrey**

### CODE REVIEW

File: opentech/apply/dashboard/templates/dashboard/applicant_dashboard.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{% load wagtailcore_tags %}
+
+{% block title %}Submission Dashboard{% endblock %}
+
+{% block content %}
+<div class="wrapper wrapper--breakout wrapper--admin">
+    <div class="wrapper wrapper--large wrapper--search">
+        <div>
+            <h3 class="heading heading--no-margin">Dashboard</h3>
+            <h5>An overview of active and past submissions</h5>
```
Comment (zerolab) [inline opentech/apply/dashboard/templates/dashboard/applicant_dashboard.html]: merge #137 into this?

Comment (todd-dembrey) [reply opentech/apply/dashboard/templates/dashboard/applicant_dashboard.html]: thats the plan

File: opentech/apply/dashboard/views.py [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+
+        return super().get_context_data(
+            my_active_submissions=my_active_submissions,
+            **kwargs,
+        )
+
+
+@method_decorator(login_required, name='dispatch')
+class DashboardView(View):
+    def dispatch(self, request):
```
Comment (zerolab) [inline opentech/apply/dashboard/views.py]: :+1:

File: opentech/apply/funds/models.py [hunk 3]
```diff
@@ -427,12 +429,19 @@ def build_json_order_by(field):
                 field = field[1:]
             else:
                 descending = False
-            return OrderBy(RawSQL("LOWER(form_data->>%s)", (field,)), descending=descending)
+            return OrderBy(RawSQL(f'LOWER({self.json_field}->>%s)', (field,)), descending=descending)
```
Comment (zerolab) [inline opentech/apply/funds/models.py]: Should this care for an empty `json_field` ?

File: opentech/apply/funds/workflow.py [hunk 4]
```diff
@@ -377,3 +380,24 @@ class DoubleStage(Workflow):
...
+
+    for phase in itertools.chain(SingleStage([None]), DoubleStage([None, None])):
+        try:
+            add_if_active(phase)
+        except AttributeError:
+            # it is actually a step
+            step = phase
+            for phase in step.phases:
+                add_if_active(phase)
+    return active
```
Comment (zerolab) [inline opentech/apply/funds/workflow.py]: a quick test?

File: opentech/apply/funds/tables.py [hunk 5]
```diff
@@ -14,25 +14,35 @@
...
 
     class Meta:
         model = ApplicationSubmission
         fields = ('title', 'status_name', 'stage', 'page', 'round', 'submit_time')
-        sequence = ('title', 'status_name', 'stage', 'page', 'round', 'lead', 'submit_time')
+        sequence = ('title', 'status_name', 'stage', 'page', 'round', 'submit_time')
         template = 'funds/tables/table.html'
+        row_attrs = {
+            'class': lambda record: '' if record.active else 'is-inactive'
+        }
```
Comment (zerolab) [inline opentech/apply/funds/tables.py]: much nicer 👍

Comment (todd-dembrey) [reply opentech/apply/funds/tables.py]: It is available on dev. But you need to login as an applicant now.

**Comment by todd-dembrey** [comment]
@chris-lawton What is up with the conflicts?

**Comment by chris-lawton** [comment]
@todd-dembrey - fixed

**Pull request merged**